### PR TITLE
Update PR template with communicating changes section

### PR DESCRIPTION
**Pull request by vaidehijoshi**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

A recent [incident](https://forem.team/vaidehijoshi/communicating-support-tooling-changes-53pm) prompted me to open this PR, which updates our PR template with a `How will this change be communicated?` section, rolled up with our `Added to documentation?` section. Not only do I think it will help us in communicating things _internally_, but I think it will make us more self-aware as a team as to when + how we need to communicate changes to the broader community as well as our users (like [this recent example](https://github.com/forem/forem/pull/12502#pullrequestreview-582866986)).

## Related Tickets & Documents

https://forem.team/vaidehijoshi/communicating-support-tooling-changes-53pm

## QA Instructions, Screenshots, Recordings

None :)

### UI accessibility concerns?

None :)

## Added tests?

- [ ] Yes
- [x] No, and this is why: _just a PR template change!_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mstruve) [inline .github/PULL_REQUEST_TEMPLATE.md:52]: Given this template is so large already and this feels very similar to `## Added to documentation?` I think we should look at combining them into a single section.

Comment (vaidehijoshi) [reply .github/PULL_REQUEST_TEMPLATE.md]: That's a good point! I've updated this -- latest commit rolls the documentation section into the new one that I have added :)

**Review by Zhao-Andy — COMMENTED** [review]
I like this update, but I'm thinking about this in the external contributor's perspective. IMO, an external contributor will most likely try to update the docs somehow or check the last option (need help on communicating). So I'm wondering now, does this seem too much of a barrier for them? If so, should we make this section for the internal team to fill out?

For our internal team, I think this update gives a helpful reminder! It's definitely a good supplement for having more documentation/process around who/when to communicate new changes.

**Comment by nickytonline** [comment]
> I like this update, but I'm thinking about this in the external contributor's perspective. IMO, an external contributor will most likely try to update the docs somehow or check the last option (need help on communicating). So I'm wondering now, does this seem too much of a barrier for them? If so, should we make this section for the internal team to fill out?
> 
> For our internal team, I think this update gives a helpful reminder! It's definitely a good supplement for having more documentation/process around who/when to communicate new changes.

I agree with @Zhao-Andy here. This really is just for us. I think it would make sense as a separate section as Andy mentions.

Having said that, we already use the dev-headsup channel for this minus the communicating this to users via the changelog, but Christina writes the changelog and posts that on forem.dev, so I wonder if this change is even required to the PR template.

The other thing to take into account is some external contributors that I've worked with already find the PR template intimidating as it is quite long.

**Comment by vaidehijoshi** [comment]
> > I like this update, but I'm thinking about this in the external contributor's perspective. IMO, an external contributor will most likely try to update the docs somehow or check the last option (need help on communicating). So I'm wondering now, does this seem too much of a barrier for them? If so, should we make this section for the internal team to fill out?
> > For our internal team, I think this update gives a helpful reminder! It's definitely a good supplement for having more documentation/process around who/when to communicate new changes.
> 
> I agree with @Zhao-Andy here. This really is just for us. I think it would make sense as a separate section as Andy mentions.

@Zhao-Andy + @nickytonline I can see the use case for making this Forem core team specific! I've made it clear that this section applies only to the Forem core team in [the latest commit](https://github.com/forem/forem/pull/12603/commits/d376b2f2d52329756d15855a00a3f106f7e1bea4). One thing to note here is that if we make this core team specific, then we need to ensure that the Forem core team member who _merges open source PRs_ takes a second to look at this section before merging. I can certainly see scenarios where something is fixed by a contributor that might impact the workflow of a developer, designer, or support team member, and we will need to put the onus of communication on the Forem core team member who shepherds an open source PR to communicate that the to the right team/department.


> Having said that, we already use the dev-headsup channel for this minus the communicating this to users via the changelog, but Christina writes the changelog and posts that on forem.dev, so I wonder if this change is even required to the PR template.


@nickytonline While we do have the dev-headsup channel, the onus is completely on the developer to remember to communicate changes. Adding this to the template as a first-class citizen (similar to our "needs tests" or "post deployment tasks" sections) could serve as a reminder to core team engineers to think about whether something should be communicated. Maybe there are things that should be previously been communicated in dev-heads up that haven't been, and we just don't know about them 😄. We also don't have a good, clear path for communicating changes to the support team right now (which is what prompted me to write that incident retro and open this PR template change), so thinking _beyond_ the engineering team and to other departments and to our users, I think there are some changes that _ought_ to be communicated that can easily slip through the cracks. For example, this PR prompted us to write a [forem.dev post to sunset a feature](https://github.com/forem/forem/pull/12502#pullrequestreview-582866986).

I know that @cmgorton writes Changelog posts, but something to consider is that putting the onus on the engineer who implements a change to flag something as "Changelog-worthy" might actually make the Changelog writing process a bit easier. I could even see us adding a `include in changelog` label or something that makes it very clear and obvious as to which PRs we want to share in a Changelog update. Down the road, if/when we start seriously working towards Forem versioning, this could be helpful in compiling + sharing which user-facing updates were released between one version of Forem versus another.

> The other thing to take into account is some external contributors that I've worked with already find the PR template intimidating as it is quite long.

Hopefully making this "forem core team only" will help alleviate this! I think combining with the preexisting documentation section per @mstruve's suggestion) will also help here :)

**Review by Zhao-Andy — APPROVED** [review]
Thanks for updating it! I like this step :)

**Review by cmgorton — APPROVED** [review]
Thanks for this Vaidehi. I can see how this will be useful for many departments at Forem and for users.

**Comment by nickytonline** [comment]
Thanks for updating it @vaidehijoshi.

**Review by nickytonline — APPROVED** [review]
🚀

**Pull request merged**
